semgrep-core/src/engine/Match_rules.ml

```diff
@@ -866,19 +872,21 @@ and match_range_with_pattern env r mvar opt_lang formula =
   | Some mval -> (
       let mval_range = MV.range_of_mvalue mval in
       match (opt_lang, mval) with
-      | Some lang, MV.Text (content, _)
-      | Some lang, MV.E (G.L (G.String (content, _))) ->
+      | Some lang, MV.Text (content, tok)
+      | Some lang, MV.E (G.L (G.String (content, tok))) ->
           (* The matched text will be interpreted according to `lang'. *)
           let lazy_ast_and_errors =
-            lazy_ast_of_string lang content (fun () ->
+            lazy_ast_of_string lang tok content (fun () ->
                 spf
                   "rule %s: metavariable-pattern: failed to fully parse the \
                    content of %s"
                   env.rule.id mvar)
           in
-          eval_nested_formula env env.xlang formula lazy_ast_and_errors
+          eval_nested_formula env
```

Comment:
I'm not clear on how this works, but would it make more sense to update `env`, and always use `env.xlang` instead of passing through both `env` and `env.xlang` in most cases?

Reply:
I agree!